buffer/src/main/java/io/netty/buffer/PooledSlicedByteBuf.java

```diff
@@ -39,20 +39,26 @@ protected PooledSlicedByteBuf newObject(Handle<PooledSlicedByteBuf> handle) {
         }
     };
 
-    static PooledSlicedByteBuf newInstance(AbstractByteBuf buffer, int index, int length, int adjustment) {
-        if (isOutOfBounds(index, length, buffer.capacity())) {
-            throw new IndexOutOfBoundsException(buffer + ".slice(" + index + ", " + length + ')');
+    static PooledSlicedByteBuf newInstance(AbstractByteBuf unwrapped, ByteBuf wrapped,
+                                           int index, int length) {
+        if (isOutOfBounds(index, length, unwrapped.capacity())) {
+            throw new IndexOutOfBoundsException(unwrapped + ".slice(" + index + ", " + length + ')');
         }
```

Comment:
can you clarify what you mean? do you mean make a function to share code for the `isOutOfBounds`/throw?